frontend/src/concepts/pipelines/content/createRun/RunPage.tsx

```diff
@@ -19,8 +20,9 @@ type RunPageProps = {
 
 const RunPage: React.FC<RunPageProps> = ({ cloneRun, contextPath }) => {
   const { namespace } = usePipelinesAPI();
+  const location = useLocation();
```

Comment:
Only change I would have done if I did it was put this inside the useRunFormData as it's not needed to be out here -- and it is directly in use for building the form data... but that's a minor thing. Personal preference I guess -- sharing so you just have another perspective.

Reply:
that makes sense. i was looking at useRunFormData as a normal function and not a hook. hence didn't put a hook inside it. but i agree it would be cleaner. and possibly just passing the id for the pipeline and then fetching the whole pipeline once inside useRunFormData